dev/src/lobster/flat_gltf.h

```diff
@@ -0,0 +1,789 @@
+            //   element."
+            if (bStride == 0 && a.type == "VEC3") {
+              // 3 components * 4 bytes
+              bStride = 12;
+            } else if (bStride == 0 && a.type == "VEC2") {
+              // 2 components * 4 bytes
+              bStride = 8;
+            }
+            // we only use floats for vertex attributes
+            if (a.componentType != 5126)
+              count = 0;
+            if (count > (int)verts.size())
+              verts.resize(count);
+            // create a GLfloat array from the vertex attribute buffer offset
+            GLfloat *vBuf = reinterpret_cast<GLfloat *>(
+                &buffers[bv.buffer][bv.offset + a.offset]);
+            _VERBOUT_ << "Buffer: " << bv.buffer << std::endl;
+            _VERBOUT_ << "Byte offset: " << (bv.offset + a.offset) << std::endl;
+            _VERBOUT_ << "Byte stride: " << bStride << std::endl;
+            _VERBOUT_ << count << " vertex attribute vectors" << std::endl;
+            for (int i = 0; i < count; i++) {
+              GLfloat *vDat = &vBuf[bStride / 4 * i];
+              switch (atType) {
+              case 0:
+                verts[i].Position[0] = vDat[0];
+                verts[i].Position[1] = vDat[1];
+                verts[i].Position[2] = vDat[2];
+                break;
+              case 1:
+                verts[i].Normal[0] = vDat[0];
+                verts[i].Normal[1] = vDat[1];
+                verts[i].Normal[2] = vDat[2];
+                break;
+              case 2: // tangents are not supported by the phong_anim shader
+                //verts[i].Tangent[0] = vDat[0];
+                //verts[i].Tangent[1] = vDat[1];
+                //verts[i].Tangent[2] = vDat[2];
+                break;
+              case 3:
+                verts[i].UV[0] = vDat[0];
+                // OpenGL UVs are vertically flipped, glTF UVs work like Vulkan UVs
+                verts[i].UV[1] = vDat[1] * -1.0f + 1.0f;
+              }
+            }
+            _VERBOUT_ << std::endl;
+          }
+          //}else{
+          //	_ERROUT_ << "Primitive does not have vertex attributes." <<
+          //std::endl;
+          //}
+
+          // TODO: really need to verify this
+          // if( p["indices"].IsNumeric() ){
+          Accessor a = accessors[p->indices()];
+          int indType = a.componentType;
+          int numInds = a.count;
+          BufferView bv = bufferviews[a.bufferView];
+          int bStride = bv.stride;
+          // create an untyped char array from the index buffer offset, to be
+          // converted to a type
+          char *indBuf = &buffers[bv.buffer][bv.offset + a.offset];
+          for (int i = 0; i < numInds; i++) {
+            fgl::Index index;
+            // get a value based on the data type indType, the array index i,
+            // and the byte stride bStride
+            switch (indType) {
+            case 5120:
```

Comment:
I'm sure we can find the constant for this..

Reply:
Comments added.